### PR TITLE
Remove exoplayer deprecations and bump Exoplayer

### PR DESCRIPTION
**Pull request by benoitdion**
#### Update the changelog
[] Update changelog

#### Describe the changes
- Bring the basic example back to a runnable state
- Bump Exoplayer to 2.10.4
- Remove deprecated usages of Exoplayer methods
- Add `ReactExoplayerConfig` as extension points to configure the Exoplayer instance

#### Related PRs
- #1653
- #1722

### CODE REVIEW

File: android-exoplayer/src/main/java/com/brentvatne/react/ReactVideoPackage.java [hunk 1]
```diff
@@ -12,19 +14,31 @@
...
     }
 
-    // Deprecated RN 0.47	
-    public List<Class<? extends JavaScriptModule>> createJSModules() {	
-        return Collections.emptyList();	
-    }	
+    // Deprecated RN 0.47
+    public List<Class<? extends JavaScriptModule>> createJSModules() {
+        return Collections.emptyList();
+    }
```
Comment (ethanperez) [inline android-exoplayer/src/main/java/com/brentvatne/react/ReactVideoPackage.java:34]: Since v5.0 supports RN 0.60+, I think it's safe to remove this

File: examples/basic/metro.config.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
```
Comment (CHaNGeTe) [inline examples/basic/metro.config.js:1]: why is this needed?

File: examples/basic/android/app/build.gradle [hunk 3]
```diff
@@ -132,8 +136,8 @@ android {
 
 dependencies {
     implementation project(':react-native-video')
-    implementation "androidx.appcompat:appcompat:1.0.0"
     implementation "com.facebook.react:react-native:+"  // From node_modules
+    implementation 'org.webkit:android-jsc:+'
```
Comment (CHaNGeTe) [inline examples/basic/android/app/build.gradle:140]: where is it used?

**Review by CHaNGeTe — APPROVED** [review]

File: examples/basic/metro.config.js (hunk 2 above)
Comment (benoitdion) [reply examples/basic/metro.config.js:1]: this is part of the RN 0.60 template. I'm hoping to simplify some of this in the future but was trying to do the minimum possible to get the example running.

File: examples/basic/android/app/build.gradle (hunk 3 above)
Comment (benoitdion) [reply examples/basic/android/app/build.gradle:140]: this is part of the RN 0.60 template. I'm hoping to simplify some of this in the future but was trying to do the minimum possible to get the example running.
